runtime/Java/src/org/antlr/v4/runtime/ParserInterpreter.java

```diff
@@ -100,6 +99,16 @@
 	protected int overrideDecision = -1;
 	protected int overrideDecisionInputIndex = -1;
 	protected int overrideDecisionAlt = -1;
+	protected boolean overrideDecisionReached = false; // latch and only override once; error might trigger infinite loop
```

Comment:
:bulb: This can be initialized to `true` if there is no override decision set, which would save some comparison operations during `visitDecision`.